marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/MintBurn.hs

```diff
@@ -313,103 +338,98 @@ rewind = H.property $ do
       expected = filter isBeforeRollback events
   equalSet expected (MintBurn.fromRows queryResult)
 
--- See Note [cardano-testnet update] on why the code was commented out.
-
 {- | Start testnet, start mint/burn indexer on it, create a single
  mint event, put it in a transaction and submit it, find the
  generated event passed back through the indexer.
 -}
+endToEnd :: Property
+endToEnd = H.withShrinks 0 $ H.propertyOnce $ (liftIO Help.setDarwinTmpdir >>) $ H.runFinallies $ H.workspace "." $ \tempPath -> do
+  (localNodeConnectInfo, networkId, socketPath) <- Help.startTestnet tempPath
+
+  -- This is the channel we wait on to know if the event has been indexed
+  indexerChan <- liftIO IO.newChan
+  -- Start indexer
+  liftIO $ do
+    coordinator <- M.initialCoordinator 1 0
+    ch <- IO.atomically . IO.dupTChan $ M._channel coordinator
+    (loop, _indexerMVar) <-
+      M.mintBurnWorker_ 123 (IO.writeChan indexerChan) Nothing coordinator ch (tempPath </> "db.db")
+    void $ IO.async loop
+    -- Receive ChainSyncEvents and pass them on to indexer's channel
+    void $ IO.async $ do
+      let chainPoint = C.ChainPointAtGenesis :: C.ChainPoint
+      c <- defaultConfigStdout
+      withTrace c "marconi" $ \trace ->
+        let indexerWorker = withChainSyncBlockEventStream socketPath networkId [chainPoint] $
+              S.mapM_ $
+                \chainSyncEvent ->
+                  IO.atomically $ IO.writeTChan ch chainSyncEvent
+            handleException NoIntersectionFound =
+              logError trace $
+                renderStrict $
+                  layoutPretty defaultLayoutOptions $
+                    "No intersection found for chain point" <+> pretty chainPoint <> "."
+         in indexerWorker `catch` handleException :: IO ()
+
+  let address :: C.Address C.ShelleyAddr
+      address = case knownAddresses !! 0 of
+        C.AddressInEra (C.ShelleyAddressInEra C.ShelleyBasedEraBabbage) addr -> addr
+        _ -> error "Invalid known address"
```

Comment:
I don't suppose `knownAddresses` would ever be empty, and this is only test code, but it always give me a jolt when I see `!!`.

```suggestion
      address = case knownAddresses of
        C.AddressInEra (C.ShelleyAddressInEra C.ShelleyBasedEraBabbage) addr : _ -> addr
        _ -> error "Invalid known address"
```